test/unit/deployment/test_extract_validator.py

```diff
@@ -0,0 +1,153 @@
+
+    def _get_values(self, first_line: str):
+        for regex, values in self.spec.items():
+            if re.match(regex, first_line, re.IGNORECASE):
+                return values() if callable(values) else values
+        LOG.warning(f"ConnectionMock.execute() called with '{first_line[:40]}...'")
+        return ()
+
+    def execute(self, *args, **kwargs):
+        statement = args[0] if len(args) else kwargs["query"]
+        first_line = statement.strip().splitlines()[0]
+        self.values = iter(self._get_values(first_line))
+        return self
+
+    def fetchone(self):
+        return next(self.values)
+
+    def fetchall(self):
+        return [ v for v in self.values ]
+
+
+class Simulator:
+    def __init__(self, nodes: int, udf_results: List[List[any]],
+                 create_script=()):
+        self.create_script = create_script
+        self.nodes = nodes
+        self.udf = Mock(side_effect=udf_results)
+        self.callback = Mock(side_effect = self._callback)
+
+    def _callback(self, n, pending):
+        LOG.debug(f"{len(pending)} of {n} nodes pending: {pending}")
+
+    @property
+    def testee(self):
+        connection = ConnectionMock({
+            r"CREATE .* SCRIPT": self.create_script,
+            r"(CREATE|DROP) ": (),
+            r"SELECT nproc\(\)": [ self.nodes ],
+            r'SELECT .*_manifest_': self.udf,
+        })
+        return ExtractValidator(
+            pyexasol_connection=Mock(execute=connection.execute),
+            timeout=timedelta(milliseconds=20),
+            interval=timedelta(milliseconds=10),
+            callback=self.callback,
+        )
+
+
+@pytest.mark.parametrize(
+    "schema, expected",
+    [
+        (None, r'"alias_manifest_[0-9]+"'),
+        ("schema", r'"schema"\."alias_manifest_[0-9]+"'),
+    ])
+def test_udf_name(schema, expected):
+    assert re.match(expected, _udf_name(schema, "alias"))
+
+
+def test_create_script_failure(archive_bucket_path):
+    create_script = Mock(side_effect=Exception("failed to create UDF script"))
+    sim = Simulator(nodes=4, udf_results=[], create_script=create_script)
+    with pytest.raises(Exception, match="failed to create UDF script") as ex:
+        assert sim.testee.verify_all_nodes("alias", "schema", archive_bucket_path)
+
+
+def test_failure(archive_bucket_path):
+    sim = Simulator(
+        nodes=4,
+        udf_results=[
+            [[1, False]],
+            [[1, False]],
+            [[1, False]],
+        ])
+    with pytest.raises(ExtractException) as ex:
+        sim.testee.verify_all_nodes("alias", "schema", archive_bucket_path)
+    assert "1 of 4 nodes are still pending. IDs: [1]" == str(ex.value)
+
+
+def test_success(archive_bucket_path):
+    sim = Simulator(
+        nodes=4,
+        udf_results=[
+            [[1, False], [2, False]],
+            [[1, True ], [2, False]],
+            [[1, True ], [2, True ]],
+        ])
+    sim.testee.verify_all_nodes("alias", "schema", archive_bucket_path)
```

Comment:
add assert

Reply:
Sorry, my bad.
`verify_all_nodes()` does not return any value.
Success is indicated by not throwing an exception.